### ISSUE TITLE
Conditional Display of Battery

### ISSUE THREAD

**Issue by boydthomson** [polybar/polybar#956]
My goals are to keep indicators hidden when they're not important.  I don't need to know how much time is remaining on my battery when it's full.  I'd like to only display the battery module when I've got < X time/percentage remaining.
  
I'm able so do something similar with the warn feature in the temperature module, but I can't find anything similar for battery.  This is how I do my cpu temp.  "label =" is blank but label-warn.  Similar 'warn' feature doesn't seem to be present in module/battery.

[module/temperature]
type = internal/temperature
thermal-zone = 0
warn-temperature = 68

format = <ramp> <label>
format-underline = ${colors.background}
format-warn = <ramp> <label-warn>
format-warn-underline = ${self.format-underline}

label =
label-warn = %temperature%
label-warn-foreground = ${colors.secondary}

**Comment by patrick96** [comment]
You can probably use `ramp-capacity` to accomplish this:
```dosini
format-discharging = <ramp-capacity> 
helper-label = My label
ramp-capacity-0 = ${self.helper-label}
ramp-capacity-1 = ${self.helper-label}
ramp-capacity-2 = ${self.helper-label}
ramp-capacity-3 = ${self.helper-label}
ramp-capacity-4 = 
ramp-capacity-5 = 
ramp-capacity-6 = 
ramp-capacity-7 = 
ramp-capacity-8 = 
ramp-capacity-9 = 
```
With this, when discharging, the module should be empty until your battery goes below 40%. Of course this is not perfect, because you will have to set the colors yourself with formatting tags.

**Comment by boydthomson** [comment]
Thanks for the suggestion.  In theory, this is exactly what I want.  The
issue is that <ramp-capacity> doesn't handle variables, so using %time% or
%percentage% in <ramp-capacity-X> or in <helper-label> displays the
variable name "%time%" instead of "4:56".  Another workaround might be to
make the module width zero for the higher ramp-capacity elements, but I
haven't been able to get that working either.

Boyd Thomson  founder / yogi / CEO / janitor
t – 604.351.6810 | stretchvancouver.com

180 E Pender St. Vancouver B.C. V6A 1T3

**Comment by patrick96** [comment]
Yes, I see. The best solution is probably to just implement `label-warn`, I'll add it to the list.

**Comment by patrick96** [comment]
Implemented in #2199
